pkg/controllers/flannelmigration/config.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flannelmigration
+
+import (
+	"fmt"
+
+	"github.com/kelseyhightower/envconfig"
+
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+//Flannel migration controller configurations
+type Config struct {
```

Comment:
Agree. I think we can improve UX trying set the default and auto detect those values. 
There are three config items which I have not put a default value to it. 

- Flannel network 
The default Flannel manifest sets it as `10.244.0.0/16`. As you said, we could set this value as our default here. Another option is to deploy a pod on one of the nodes and get this value from /run/flannel/subnet.env
```
ubuntu@ip-172-16-101-243:/run/flannel$ cat subnet.env 
FLANNEL_NETWORK=10.244.0.0/16
FLANNEL_SUBNET=10.244.0.1/24
FLANNEL_MTU=8951
FLANNEL_IPMASQ=true
```

- Flannel daemonset name. 
The default Flannel manifest sets it as `kube-flannel-ds-amd64`. Again, we can set same default value here.
Also I am thinking adding code to search for daemonset with a label `app: flannel`, if any daemonset exists, it just replace the default one. 

- Flannel MTU
This value has no default in Flannel either. It is auto detected on the interface. We have to deploy a pod to get it from subnet.env file. Once this value has been detected, we also need to update `veth_mtu` in `calico-config` ConfigMap. 

In a word, what I need to do is
- Search daemonet with label `app: flannel`, if exists, use it. Otherwise, use default value.
- Deploy a pod, read subnet.env file and set `Flannel Network` and `Flannel MTU` config. 
- Update `veth_mtu` in calico configMap. 

@caseydavenport WDTY?

Reply:
> Flannel daemonset name.

Yep. we could probably get away with just looping daemonsets in the kube-system namespace and finding the one that has "flannel' in the name. But also, I don't expect this to be modified by users so we could even just hard code for now. 

>  Another option is to deploy a pod on one of the nodes and get this value from /run/flannel/subnet.env

We don't actually need to deploy another pod, right? We can simply mount that file into the controller?

Reply:
Sounds good. I have push a commit to autodetect Flannel configs.